src/index.ts

```diff
@@ -241,7 +241,10 @@ prog
     const installSpinner = ora(Messages.installing(deps.sort())).start();
     try {
       const cmd = await getInstallCmd();
-      await execa(cmd, getInstallArgs(cmd, deps));
+      const args = getInstallArgs(cmd, deps).join(' ');
+      await shell.exec(`${cmd} ${args}`, {
+        silent: true,
+      });
```

Comment:
@agilgur5 `await execa(cmd, getInstallArgs(cmd, deps), { shell: true })` does not work

Reply:
@agilgur5 Any update? Can this go in or what changes should I make? Would rather not maintain a fork of TSDX

Reply:
Please don't expect an immediate response only 24 hours after you took 5 days to respond to a comment... I pay companies for premium support and don't get a response that fast, but you're asking a volunteer to respond faster than that and than you yourself did, for a private registry feature at that.
That's counterproductive and does the opposite of motivating maintainers.

> Would rather not maintain a fork of TSDX

It sounds like you already are if you have custom templates??? `tsdx create` does not create an `.npmrc` and all deps it installs are public, so there's no need for one.

We have also generally recommend [`patch-package`](https://github.com/ds300/patch-package) when one needs further customization and this is a good use-case for it

Reply:
It sounds like this is an upstream issue in `execa` and so I think an issue should be filed there. I looked there and there actually is an existing issue on this exact topic: https://github.com/sindresorhus/execa/issues/380

The fix there is to use `cwd`. If that resolves it, I think it would be a much better option than using `shelljs` which has various explicit caveats to its usage (several listed in `execa`'s `shell` option that I linked previously)